archive/frames/models.py

```diff
@@ -159,6 +162,62 @@ def as_dict(self):
         ret_dict['related_frames'] = list(self.related_frames.all().values_list('id', flat=True))
         return ret_dict
 
+
+class Thumbnail(models.Model):
+    frame = models.ForeignKey(
+        Frame, 
+        on_delete=models.CASCADE, 
+        related_name='thumbnails', 
+        help_text="The frame this thumbnail is associated with"
+    )
+    size = models.CharField(
+        max_length=20,
+        help_text="String description of the size of the thumbnail"
+    )
+    basename = models.CharField(
+        max_length=1000, 
+        db_index=True, 
+        unique=True,
+        help_text="The basename of the thumbnail"
+    )
+    extension = models.CharField(
+        max_length=20,
+        help_text="The file extension of the thumbnail"
+    )
+    key = models.CharField(
+        max_length=32,
+        help_text="The key used to store the thumbnail in the file store",
+        default=''
+    )
+
+    def as_dict(self):
+        ret_dict = model_to_dict(self)
+        ret_dict['url'] = self.url
+        ret_dict['size'] = self.size
+        return ret_dict
+    
+    @property
+    def filename(self):
+        """
+        Returns the full filename for the thumbnail
+        """
+        return '{0}{1}'.format(self.basename, self.extension)
+     
+    @cached_property
+    def url(self):
+        path = ThumbnailFile.get_filestore_path_from_frame_metadata(self.frame.site_id, self.frame.instrument_id, 
```

Comment:
@jnation3406 you left a comment about the previous version of this function needing the header to construct the filestore path using the DataFile class. Problem was, we needed to construct the DataFile class with all valid metadata so it could determine the filestore path using only part of that metadata, which as you mentioned, would lead to unnecessary table joins and accesses. So what I've done in [this commit ](https://github.com/observatorycontrolsystem/ocs_archive/pull/11/commits/0d8ccecd17ab8ad840be092065e65622e7ed080f#diff-8e1ac2eeaa8f8d1f99c44a4fe2095cbf4cbdf2cbfeee42689c1903138e172786) was add a static method to the ThumbnailFile class to take in _only_ the metadata needed to construct the filestore path. To preserve the DataFile interface (subclasses of this need implement `get_filestore_path` for the filestore to work), I just call the static method and use the data stored in the ThumbnailFile's metadata.

This allows the science archive to provide a minimal set of metadata without having to do too much DB work.

Reply:
This is okay I guess, but if you want to avoid importing the ThumbnailFile directly, you could keep using the get_file_store_path util function but instead of passing the header, just pass a metadata dict of the header keys you need, i.e. 
`get_file_store_path(self.filename, {'site_id': self.frame.site_id, 'instrument_id': self.frame.instrument_id, 'observation_day': self.frame.observation_day.strftime('%Y%m%d')})`